script.js

```diff
@@ -1,4 +1,401 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
+var numOfTimesUserWon = 0;
+var numOfTimesPcWon = 0;
+var numOfDraws = 0;
+var numOfGamesPlayed = 0;
+var pcWinningPercent = 0;
+var userWinningPercent = 0;
+var userName = "";
+var gameMode = "Waiting for UserName";
+var gameVersion = "Normal Version";
+var reverseCount = 0;
+var recentWinner = "";
+
+var main = function (userInput) {
+  var myOutputValue = "";
+
+  // If game mode is "Waiting for UserName", store input as user's name into a variable
+  if (gameMode == "Waiting for UserName") {
+    userName = userInput;
+    myOutputValue =
+      `Welcome ${userName}! 👋<br ><br >` +
+      `Select the version that you would like to play by submitting "Normal" or "Korean".`;
+    gameMode = "Version Selection";
+    return myOutputValue;
+  } else if (gameMode == "Version Selection") {
+    // If game mode is in "Version Selection" and no issue with input validation, show user the game instructions
+    var resultOfValidityCheck = validityCheck(userInput);
+    if (resultOfValidityCheck == true) {
+      return introduceGame(userInput);
+    }
+    return resultOfValidityCheck;
+  } else {
+    // If game mode is not "Waiting for UserName" or "Version Selection", run
+    myOutputValue = generateWinner(userInput);
+    return myOutputValue;
+  }
+};
+
+// ===== Function to introduce game features =====
+
+var introduceGame = function (userInput) {
+  var gameInstructions = "";
+  gameMode = "Original Mode";
+  // If user select Normal version, set game version to Normal
+  if (userInput == "Normal") {
+    gameVersion = "Normal Version";
+    gameInstructions =
+      `You have selected the ${gameVersion}.<br ><br >` +
+      `Submit "scissors", "paper", "stone", "reversed scissors", "reversed paper" or "reversed stone" to start playing!<br ><br >` +
+      `You can also submit "reverse" to reverse the game rules.`;
+    return gameInstructions;
+  }
+  // If user select Korean version, set game version to Korean
+  if (userInput == "Korean") {
+    gameVersion = "Korean Version";
+    gameInstructions =
+      `You have selected the ${gameVersion}.<br ><br >` +
+      `Submit "scissors", "paper", "stone", "reversed scissors", "reversed paper" or "reversed stone" to start playing!<br ><br >` +
+      `You can also submit "reverse" to reverse the game rules.`;
+    return gameInstructions;
```

Comment:
since both the code in these cases remains largely the same except for lines 45 and 54. You can actually just combine these two into one statement to reduce redundancy. If you see that your code seems to copy some code with minimal changes, the rule of thumb is that you can probably reduce the redundancy in some way